lib/practice_exercises.rb

```diff
@@ -1,13 +1,57 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(1)
 def remove_duplicates(list)
-  raise NotImplementedError, "Not implemented yet"
+  i = 0
+  list.length.times do
+    if list[i] == list[i + 1]
+      list.delete_at(i)
+      i += 1
+    else
+      i += 1
+    end
+  end
+  return list
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: although there is a loop within a loop, I don't think it's quadratic 
```

Comment:
Actually since you have an outer loop which runs n times (the length of the array) and an inner loop that **also** runs n times, you have an O(n<sup>2</sup>) algorithm here.